libraries/Adafruit_LittleFS/src/Adafruit_LittleFS_File.cpp

```diff
@@ -319,24 +331,17 @@ bool File::_isOpen(void)
 //            suddenly (unexpectedly?) have different values.
 char const* File::name (void)
 {
-  this->_LockFilesystem();
-  char const* ret = this->_name;
-  this->_UnlockFilesystem();
-  return ret;
+  return this->_name;
```

Comment:
@henrygab in my defense to skip FS lock, we only guard again lfs_* call since lfs is not reentrant for now. Over time, lfs will follow fatfs to provide option/hook for locking. By that we can revert most of this implementation by using the lfs lock hook. 

The scenario when a same File object is used by user to write as file in one thread, but close and open as directory (or other filepath) in aother thread  etc .. should be guarded by application itself.

Reply:
OK.  It's a design choice, and I admit I was **_very_** conservative in the locking.  Your changes will allow two File objects to get a subset of metadata without locking:
* `isOpen()`
* `name()`
* `isDirectory()`

each of the above can be returned without accessing LFS, so this appears safe.